web/server/codechecker_server/api/authentication.py

```diff
@@ -93,9 +97,63 @@ def getLoggedInUser(self):
         else:
             return ""
 
+    @timeit 
+    def createLinkGithub(self):
+        """
+        This functin is for creating a autehntication link for OAuth for Github.
+        """
+        self.oauth_config = self.oauth_config_github
+
+        if not self.oauth_config.get("enabled"):
+            raise codechecker_api_shared.ttypes.RequestFailed(
+                codechecker_api_shared.ttypes.ErrorCode.AUTH_DENIED,
+                "OAuth authentication is not enabled.11")
+        
+        client_id = self.oauth_config["oauth_client_id"]
+        client_secret = self.oauth_config["oauth_client_secret"]
+        scope = self.oauth_config["oauth_scope"]
+        authorization_uri = self.oauth_config["oauth_authorization_uri"]
+
+        # Create an OAuth2Session instance
+        session = OAuth2Session(client_id, client_secret, scope=scope)
+        url, _ = session.create_authorization_url(authorization_uri)
+        
+        return url
+
+    @timeit
+    def createLinkGoogle(self):
+        """
+        This function is for creating an authentication link for OAuth for Google.
```

Comment:
This code is almost a duplicate of `createLinkGithub`. This is why I asked to simplify the API.

Reply:
fixed